samples/nrf_desktop/src/hw_interface/buttons_matrix.c

```diff
@@ -164,14 +249,30 @@ static void matrix_scan_fn(struct k_work *work)
 		}
 
 		/* Make sure that mode is set before callbacks are enabled */
-		atomic_set(&scanning, false);
+		int err = 0;
+
+		k_spinlock_key_t key = k_spin_lock(&lock);
+		switch (state) {
+		case STATE_SCANNING:
+			state = STATE_ACTIVE;
+			err = callback_ctrl(true);
+			break;
+
+		case STATE_SUSPENDING:
+			state = STATE_ACTIVE;
+			err = suspend();
+			break;
+
+		default:
+			__ASSERT_NO_MSG(false);
+			break;
+		}
+		k_spin_unlock(&lock, key);
 
-		int err = callback_ctrl(true);
 		if (err) {
 			LOG_ERR("cannot enable callbacks");
```

Comment:
When is STATE_SUSPENDING, user will get this error message if the `err = suspend()` returns non-zero.

Reply:
I moved error handling as discussed offline.